x-pack/plugins/cloud_security_posture/server/routes/benchmarks/utilities.ts

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedObjectsClientContract } from '@kbn/core-saved-objects-api-server';
+import { QueryDslQueryContainer } from '@kbn/data-views-plugin/common/types';
+import { AgentPolicy, PackagePolicy } from '@kbn/fleet-plugin/common';
+import { MappingRuntimeFields } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import {
+  CSP_RULE_TEMPLATE_SAVED_OBJECT_TYPE,
+  LATEST_FINDINGS_INDEX_DEFAULT_NS,
+} from '../../../common/constants';
+import { getSafePostureTypeRuntimeMapping } from '../../../common/runtime_mappings/get_safe_posture_type_runtime_mapping';
+import { CspRuleTemplate } from '../../../common/schemas';
+import { BenchmarkId, Benchmark, BenchmarkVersion2 } from '../../../common/types';
+import {
+  getBenchmarkFilter,
+  isNonNullable,
+  getBenchmarkFromPackagePolicy,
+} from '../../../common/utils/helpers';
+import { AgentStatusByAgentPolicyMap } from '../../lib/fleet_util';
+import { getClusters } from '../compliance_dashboard/get_clusters';
+import { getStats } from '../compliance_dashboard/get_stats';
+
+export const getRulesCountForPolicy = async (
+  soClient: SavedObjectsClientContract,
+  benchmarkId: BenchmarkId
+): Promise<number> => {
+  const rules = await soClient.find<CspRuleTemplate>({
+    type: CSP_RULE_TEMPLATE_SAVED_OBJECT_TYPE,
+    filter: getBenchmarkFilter(benchmarkId),
+    perPage: 0,
+  });
+
+  return rules.total;
+};
+
+export const createBenchmarks = (
+  soClient: SavedObjectsClientContract,
+  agentPolicies: AgentPolicy[],
+  agentStatusByAgentPolicyId: AgentStatusByAgentPolicyMap,
+  cspPackagePolicies: PackagePolicy[]
+): Promise<Benchmark[]> => {
+  const cspPackagePoliciesMap = new Map(
+    cspPackagePolicies.map((packagePolicy) => [packagePolicy.id, packagePolicy])
+  );
+
+  return Promise.all(
+    agentPolicies.flatMap((agentPolicy) => {
+      const cspPackagesOnAgent =
+        agentPolicy.package_policies
+          ?.map(({ id: pckPolicyId }) => {
+            return cspPackagePoliciesMap.get(pckPolicyId);
+          })
+          .filter(isNonNullable) ?? [];
+
+      const benchmarks = cspPackagesOnAgent.map(async (cspPackage) => {
+        const benchmarkId = getBenchmarkFromPackagePolicy(cspPackage.inputs);
+        const rulesCount = await getRulesCountForPolicy(soClient, benchmarkId);
+        const agentPolicyStatus = {
+          id: agentPolicy.id,
+          name: agentPolicy.name,
+          agents: agentStatusByAgentPolicyId[agentPolicy.id]?.total,
+        };
+        return {
+          package_policy: cspPackage,
+          agent_policy: agentPolicyStatus,
+          rules_count: rulesCount,
+        };
+      });
+
+      return benchmarks;
+    })
+  );
+};
+
+export const createBenchmarksV2 = async (
```

Comment:
this is not a utility function, it is part of v2 implementation, should be moved to v2.ts

besides that, rename the function, it is missleading. Since it doesn't create benchmarks.